src/core/geometry/qgspointv2.cpp

```diff
@@ -462,3 +462,23 @@ double QgsPointV2::azimuth( const QgsPointV2& other ) const
   double dy = other.y() - mY;
   return ( atan2( dx, dy ) * 180.0 / M_PI );
 }
+
+
+QgsPointV2 QgsPointV2::project( double distance, double azimuth, double inclinaison ) const
+{
+    double rads_xy = azimuth * M_PI / 180.0;
+    if ( !is3D() )
```

Comment:
This looks unexpected to me. If I project a 2D point using an inclination I'd expect a 3D point result projected from an initial 0 z value.